### PR TITLE
take share target into account when updating recipient etags

### LINKED ISSUE #16505: 8.1 beta 2 : share_folder / sync / etag propagation
My test focus on shared file synchronisation with desktop client, and it failed.

I use the share_folder parameter in server file config.php to define an alternate folder (not /) :
  'share_folder' => '/share_subdir',

Steps to reproduce :
1. create two users "u1" and "u2"
2. configure desktop for user "u2"
3. login as "u1"
4. create or upload a file named "file1"
5. share "file1" with user "u2"
6. check in the desktop if the file appear in "share_subdir/file1"

Expected result :
- shared files should be synchronized in "share_folder" on desktop client

Actual result :
- files are not synchronized until other modifications are made in share_folder
- files appears in web interface for user u2

I will post cadaver result soon.

Antoine

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes #16505

To test:
- change the config value of `'share_folder'` to something other than `'/'`
- create two users user1 and user2
- check the etag of the folder defined as `share_folder`
- share a file from user1 to user2
- check if the etag of the `share_folder` has changed

cc @DeepDiver1975 @PVince81

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Thanks a lot!  Any chance to get this unit tested?

**Comment by PVince81** [comment]
It doesn't seem to work. In your steps you didn't specify to manually create the shared folder.

Here are my steps:

1. Change the config value of `share_folder` to `my_shares`
1. Create two users "user1" and "user2"
1. Login as "user1"
1. Create a folder "my_shares"
1. With cadaver as "user1", do a propget on "my_shares": `DAV: getetag = "55895360e1c98"`
1. Login as "root" in the web UI
1. Create a folder "stuff"
1. Share "stuff" with user1 and user2
1. With cadaver as "user1", do a propget on "my_shares": `DAV: getetag = "55895360e1c98"`

Note: the etag of the root folder for user1 and user2 did change.

**Comment by DeepDiver1975** [comment]
@icewind1991 can you please take a look? THX

**Comment by icewind1991** [comment]
>  In your steps you didn't specify to manually create the shared folder.

That shouldn't be needed

I re-tested it by sharing and un-sharing a few files from user1 to user2 and the etag for user2 changes every time

**Comment by PVince81** [comment]
Then step3 is impossible. I cannot check the etag of a folder that does not exist.
Can you compare in what way my steps might differ from yours ?

**Comment by PVince81** [comment]
Pffff, I was probably on the wrong branch when testing. Sorry.

I tried again with two cases: one without pre-creating "my_shares" and one with pre-creating it.
In both cases the etag of that folder changes for the recipient, for file shares and folder shares.

:+1:

**Comment by PVince81** [comment]
@DeepDiver1975 @nickvergessen @schiesbn your seal of approval ?

**Comment by PVince81** [comment]
Wait a minute, there is no unit test.

- [ ] TODO: add unit test

**Comment by PVince81** [comment]
Unit test added here https://github.com/owncloud/core/commit/56d9df34cb4bed8267453bcf894f7ead775a5827

@icewind1991 please approve it.

My thumbs up still applies for the first commit.

**Comment by icewind1991** [comment]
:+1: thanks for taking care of the unit tests

**Comment by PVince81** [comment]
This PR requires another seal of approval

File: apps/files_sharing/lib/updater.php [hunk 1]
```diff
@@ -77,10 +77,10 @@ static public function postShareHook($params) {
 			$shareType = $params['shareType'];
 
 			if ($shareType === \OCP\Share::SHARE_TYPE_USER) {
-				self::correctUsersFolder($shareWith, '/');
+				self::correctUsersFolder($shareWith, $params['fileTarget']);
```
Comment (rullzer) [inline apps/files_sharing/lib/updater.php]: I hate those arrays that are passed to functions... but not your fault ;)

**Comment by rullzer** [comment]
Code looks good. And my tests with cadaver show correct behaviour. :+1:

**Comment by PVince81** [comment]
Thanks a lot for the review @rullzer, merging.

**Comment by PVince81** [comment]
@karlitschek Backport to stable8 ? (I checked, there are no conflicts)

**Comment by karlitschek** [comment]
backport is fine :+1:

**Comment by MorrisJobke** [comment]
@PVince81 @icewind1991 @karlitschek Is a backport to stable7 possible?

**Comment by karlitschek** [comment]
@MorrisJobke Backport makes sense. Thanks!

**Comment by cdamken** [comment]
Thanks @karlitschek 

@MorrisJobke 

00003741
00003709

**Pull request merged**